lib/pipelineTemplateVersionFactory.js

```diff
@@ -130,6 +130,47 @@ class PipelineTemplateVersionFactory extends BaseFactory {
         return newPipelineTemplateVersion;
     }
 
+    /**
+     * List all the versions of a pipeline template
+     * @method create
+     * @param  {Object}     config               Config object
+     * @param templateMetaFactory
+     * @param  {Datastore}  config.datastore     Object that will perform operations on the datastore
+     * @param  {String}     config.params.name          The template name
+     * @param  {String}     config.params.namespace     The template namespace
+     * @param  {String}     config.sort                 The sort order of the list
+     * @param  {Object}     config.paginate             Pagination parameters
+     * @param  {Number}     config.paginate.count       Number of items per page
+     * @param  {Number}     config.paginate.page        Specific page of the set to return
+     * @return {Promise}                         List of PipelineTemplateVersionModel
+     */
+    async list(config, templateMetaFactory) {
+        const { name, namespace } = config.params;
+
+        if (!name || !namespace) {
+            return [];
```

Comment:
Returning empty list when only `name` or `namespace` is specified could be misleading
We should throw error indicating both `name` and `namespace` is mandatory.